packages/builder/src/stores/builder/screenComponent.ts

```diff
@@ -0,0 +1,80 @@
+import { derived } from "svelte/store"
+import { tables } from "./tables"
+import { selectedScreen } from "./screens"
+import { viewsV2 } from "./viewsV2"
+import { DerivedBudiStore } from "../BudiStore"
+import { findComponentsBySettingsType } from "@/helpers/screen"
+import { Screen, Table, ViewV2 } from "@budibase/types"
+
+interface BuilderScreenComponentStore {}
+
+interface DerivedScreenComponentStore extends BuilderScreenComponentStore {
+  errors: Record<string, string[]>
+}
+
+export class ScreenComponentStore extends DerivedBudiStore<
+  BuilderScreenComponentStore,
+  DerivedScreenComponentStore
+> {
+  constructor() {
+    const makeDerivedStore = () => {
+      return derived(
+        [selectedScreen, tables, viewsV2],
+        ([$selectedScreen, $tables, $viewsV2]): DerivedScreenComponentStore => {
+          const datasources = flattenTablesAndViews($tables.list, $viewsV2.list)
+          return {
+            errors: getInvalidDatasources($selectedScreen, datasources),
+          }
+        }
+      )
+    }
+
+    super({}, makeDerivedStore)
+  }
+}
+
+export const screenComponentStore = new ScreenComponentStore()
+
+function flattenTablesAndViews(tables: Table[], views: ViewV2[]) {
+  return {
+    ...tables.reduce(
+      (list, table) => ({
+        ...list,
+        [table._id!]: table,
+      }),
+      {}
+    ),
+    ...views.reduce(
+      (list, view) => ({
+        ...list,
+        [view.id]: view,
+      }),
+      {}
+    ),
+  }
+}
+
+function getInvalidDatasources(
+  screen: Screen,
+  datasources: Record<string, any>
+) {
+  const friendlyNameByType = {
+    table: "table",
+    view: "view",
+    viewV2: "view",
+  }
+
+  const result: Record<string, string[]> = {}
+  for (const component of findComponentsBySettingsType(screen, "table")) {
+    const { resourceId, type, label } = component.dataSource
```

Comment:
NAB: This makes the assumption that all settings of type `table` are stored under the key `dataSource`. I know that is true of the manifest today, but it's not something we can guarantee as keys are otherwise arbitrary. This also may not account for custom components which may use other keys too.

The complete solution is more complicated because you'd have to return both the component and key for set setting you find. It's not super important to do this as the current manifest satisfies this constraint, but worth keeping in mind I guess.